eras/shelley/impl/src/Cardano/Ledger/Shelley/PParams.hs

```diff
@@ -515,6 +516,28 @@ pvCanFollow _ SNothing = True
 pvCanFollow (ProtVer m n) (SJust (ProtVer m' n')) =
   (succVersion m, 0) == (Just m', n') || (m, n + 1) == (m', n')
 
+upgradeUpdate ::
+  forall era.
+  ( EraPParams era
+  , EraPParams (PreviousEra era)
+  , EraCrypto (PreviousEra era) ~ EraCrypto era
+  , UpgradePParams StrictMaybe era ~ ()
+  ) =>
+  Update (PreviousEra era) ->
+  Update era
+upgradeUpdate (Update pp epoch) = Update (upgradeProposedPPUpdates @era pp) epoch
+
+upgradeProposedPPUpdates ::
+  ( EraPParams era
+  , EraPParams (PreviousEra era)
+  , EraCrypto (PreviousEra era) ~ EraCrypto era
+  , UpgradePParams StrictMaybe era ~ ()
+  ) =>
+  ProposedPPUpdates (PreviousEra era) ->
+  ProposedPPUpdates era
+upgradeProposedPPUpdates (ProposedPPUpdates ppus) =
+  ProposedPPUpdates $ upgradePParamsUpdate () <$> ppus
```

Comment:
@lehins I generalized like this, but it feels wrong, particularly the `UpgradePParams StrictMaybe era ~ ()` .. 
Is there a better way to do it?